tests/integration/test_charm.py

```diff
@@ -152,6 +157,103 @@ async def test_jdbc_endpoint(ops_test: OpsTest, test_pod):
+    )
+
+    # Assert that both kyuubi-k8s and postgresql-k8s charms are in active state
+    assert ops_test.model.applications[APP_NAME].status == "active"
+    assert ops_test.model.applications[charm_versions.postgres.application_name].status == "active"
+
+
+@pytest.mark.abort_on_fail
+async def test_jdbc_endpoint_with_postgres_metastore(ops_test: OpsTest, test_pod, charm_versions):
+    """Test the JDBC endpoint exposed by the charm."""
+    logger.info("Running action 'get_jdbc_endpoint' on kyuubi-k8s unit...")
+    kyuubi_unit = ops_test.model.applications[APP_NAME].units[0]
+    action = await kyuubi_unit.run_action(
+        action_name="get-jdbc-endpoint",
+    )
+    result = await action.wait()
+
+    jdbc_endpoint = result.results.get("endpoint")
+    logger.info(f"JDBC endpoint: {jdbc_endpoint}")
+
+    pod_name = test_pod
+
+    logger.info(
+        "Testing JDBC endpoint by connecting with beeline" " and executing a few SQL queries..."
+    )
+    process = subprocess.run(
+        [
+            "./tests/integration/test_jdbc_endpoint.sh",
+            pod_name,
+            jdbc_endpoint,
+            "db_postgres_metastore",
+            "table_postgres_metastore",
+        ],
+        capture_output=True,
+    )
+    print("========== test_jdbc_endpoint.sh STDOUT =================")
+    print(process.stdout.decode())
+    print("========== test_jdbc_endpoint.sh STDERR =================")
+    print(process.stderr.decode())
+    logger.info(f"JDBC endpoint test returned with status {process.returncode}")
+    assert process.returncode == 0
+
+    # Fetch password for default user from postgresql-k8s
+    postgres_unit = ops_test.model.applications[charm_versions.postgres.application_name].units[0]
+    action = await postgres_unit.run_action(
+        action_name="get-password",
+    )
+    result = await action.wait()
+    password = result.results.get("password")
+
+    # Fetch host address of postgresql-k8s
+    status = await ops_test.model.get_status()
+    postgresql_host_address = status["applications"][charm_versions.postgres.application_name][
+        "units"
+    ][f"{charm_versions.postgres.application_name}/0"]["address"]
+
+    # Connect to PostgreSQL metastore database
+    connection = psycopg2.connect(
+        host=postgresql_host_address,
+        database=METASTORE_DATABASE_NAME,
+        user="operator",
+        password=password,
+    )
+
+    # Fetch number of new db and tables that have been added to metastore
+    num_dbs = num_tables = 0
+    with connection.cursor() as cursor:
+        cursor.execute(""" SELECT * FROM "DBS" WHERE "NAME" != 'default' """)
+        num_dbs = cursor.rowcount
+        cursor.execute(""" SELECT * FROM "TBLS" """)
+        num_tables = cursor.rowcount
+
+    connection.close()
+
+    # Assert that new database and tables have indeed been added to metastore
+    assert num_dbs != 0
+    assert num_tables != 0
```

Comment:
**todo** Can we check also that when we remove the relation, we fall back on using the derby hive metastore?